### PR TITLE
Enh #6132: Refactor and Remove AjaxButton

### PR DESCRIPTION
**Pull request by gevorgmansuryan**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [x] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included
- [ ] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by luke-** [comment]
@gevorgmansuryan Are you planning to refactor other occurrences of `AjaxButton` in this PR as well?

**Comment by luke-** [comment]
@gevorgmansuryan Is this PR ready to merge? If yes, can you please resolve conflicts?

**Comment by gevorgmansuryan** [comment]
> @gevorgmansuryan Is this PR ready to merge? If yes, can you please resolve conflicts?

@luke- there is another refactor which conflicts with most of changes from this PR, it's not just conflicts. will check and fix all the cases. After this PR will be ready to merge.

**Comment by gevorgmansuryan** [comment]
m

> > @gevorgmansuryan Is this PR ready to merge? If yes, can you please resolve conflicts?
> 
> @luke- there is another refactor which conflicts with most of changes from this PR, it's not just conflicts. will check and fix all the cases. After this PR will be ready to merge.

@luke- done.

**Comment by luke-** [comment]
@gevorgmansuryan Can you please check the failing tests?

**Comment by gevorgmansuryan** [comment]
> @gevorgmansuryan Can you please check the failing tests?

@luke- fixed

File: protected/humhub/modules/space/resources/js/humhub.space.js
Comment (luke-) [inline protected/humhub/modules/space/resources/js/humhub.space.js:74]: See `admin.setAsDefault` comment.

File: protected/humhub/modules/space/views/membership/requestMembership.php
Comment (luke-) [inline protected/humhub/modules/space/views/membership/requestMembership.php:60]: Do we really manually need to include the loader here. Is there no more generic approach?

Maybe you can find some generic solution here: 
https://github.com/humhub/humhub/pull/6389

**Review by luke- — CHANGES_REQUESTED** [review]
@gevorgmansuryan Thanks tests are fine now. Please see my remarks.

File: protected/humhub/modules/space/views/membership/requestMembership.php
Comment (gevorgmansuryan) [reply protected/humhub/modules/space/views/membership/requestMembership.php:60]: okay

File: protected/humhub/modules/admin/resources/js/humhub.admin.js
Comment (gevorgmansuryan) [reply protected/humhub/modules/admin/resources/js/humhub.admin.js:148]: ui.modal.load works via get method, in this case we need post (form sumbit). Better rename to moduleSetAsDefault

File: protected/humhub/modules/space/resources/js/humhub.space.js
Comment (gevorgmansuryan) [reply protected/humhub/modules/space/resources/js/humhub.space.js:74]: same here, we need post request.  With naming i think it's ok in this case, it's space.js

File: protected/humhub/modules/admin/resources/js/humhub.admin.js
Comment (luke-) [reply protected/humhub/modules/admin/resources/js/humhub.admin.js]: @gevorgmansuryan What about a `ui.modal.post` or something like that?

**Comment by luke-** [comment]
@gevorgmansuryan Please see my comment https://github.com/humhub/humhub/pull/6172#discussion_r1249757218 and it seems the tests are failing again.

**Comment by gevorgmansuryan** [comment]
@luke- tests are fixed. And about ui.modal.post: it's for loading modal content via post. our case it's about buttons in already loaded modal content, so we can't use any modal related methods here.

**Review by luke- — APPROVED** [review]

**Pull request merged**
